### ISSUE TITLE
The loss has been rising and tends to oscillate.

### ISSUE THREAD

**Issue by huhailang9012** [liyunqianggyn/Deep-Unsupervised-Image-Hashing#6]
![微信截图_20211026225019](https://user-images.githubusercontent.com/16838538/138905635-bafa84d5-f68d-4c3c-8900-88ee12db77f6.png)

At the beginning of training, the real-time loss keeps rising and tends to oscillate later. I have tried several times and it is the same. May I ask whether this is a problem with the model implementation or my training graph output.
Thank you in advance.

**Comment by liyunqianggyn** [comment]
Thanks for your interests! 
I will try to run the code this week and I'll leave another comment here when I have. Please stay tuned! :)

**Comment by liyunqianggyn** [comment]
Thanks again for your interest! 

I run the experiments with the following setting: 
"train_dataset 50000
test_dataset 10000
database_dataset 50000
Lr = 0.0001
bit = 64
gamma = 6"
Unfortunately the same loss curve occurs for my running that the cosine MSE loss first goes up then stays at a high value. 

I first think this is because the "gamma" is too large as in
![gamma](https://user-images.githubusercontent.com/29226200/139642444-b5c79941-61e0-41b8-b34a-346efec5790f.png)
that makes the training more focus on **_pushing to discrete binary value instead minimizing cosine MSE loss_**. I change the gamma to 2 and 0.5 to check the loss curve, here is some output training log:

- with default setting of gamma = 6, the MSE loss increases from 0.12 to 0.18, it can get mAP =59.7% ( in paper it is 59.5%) 
[slurm-6748051.pdf](https://github.com/liyunqianggyn/Deep-Unsupervised-Image-Hashing/files/7451638/slurm-6748051.pdf)
- with setting of gamma = 2, the MSE loss increases from 0.08 to 0.09, it can get mAP =55.6% (decreased) 
[slurm-6748705.pdf](https://github.com/liyunqianggyn/Deep-Unsupervised-Image-Hashing/files/7451751/slurm-6748705.pdf)
- with setting of gamma = 0.5, the MSE loss changes from 0.080 to 0.079, it can get mAP =53.9% (decreased) 
[slurm-6748708.pdf](https://github.com/liyunqianggyn/Deep-Unsupervised-Image-Hashing/files/7451769/slurm-6748708.pdf)

The gamma is used to balance between pushing to discrete and minimizing the MSE loss.  **_With setting a large gamma, this could increase the MSE loss. I guess this could be a reason why the MSE loss increases._** But I still can not fully explain why increasing loss (with gamma = 6,  from 0.12 to 0.18) can get better mAP increased ( from 48.5% to 59.7%). Please let me know if you have good explanation for this, thanks a lot! 
Hope it can help you! this is forcing us to rethink more :) 

**BTW: if you can not implement the performance, could you please refer to the** [issue](https://github.com/liyunqianggyn/Deep-Unsupervised-Image-Hashing/issues/5), **_Or_ refer to the** [issue1](https://github.com/swuxyj/DeepHash-pytorch/issues/20) and [issue2](https://github.com/swuxyj/DeepHash-pytorch/issues/18).
